repair/row_level.cc

```diff
@@ -2871,12 +2871,8 @@ class row_level_repair {
 future<> repair_cf_range_row_level(repair::shard_repair_task_impl& shard_task,
         sstring cf_name, table_id table_id, dht::token_range range,
         const std::vector<gms::inet_address>& all_peer_nodes) {
-    return seastar::futurize_invoke([&shard_task, cf_name = std::move(cf_name), table_id = std::move(table_id), range = std::move(range), &all_peer_nodes] () mutable {
-        auto repair = row_level_repair(shard_task, std::move(cf_name), std::move(table_id), std::move(range), all_peer_nodes);
-        return do_with(std::move(repair), [] (row_level_repair& repair) {
-            return repair.run();
-        });
-    });
+    auto repair = row_level_repair(shard_task, std::move(cf_name), std::move(table_id), std::move(range), all_peer_nodes);
+    co_return co_await repair.run();
```

Comment:
nit: the `co_return` is superfluous.

Reply:
It is not, we don't allow returning unready futures.

Reply:
Sorry @denesb, but `co_await repair.run();` will wait on `repair.run()`'s future until it's resolved before the coroutine's future is resolved, so no unready future is going to be returned.

Reply:
it's like a superfluous `return;` statement in
```c++
void foo() {
    // do stuff
    return;
}
```

Reply:
@denesb @bhalevy hi Botond and Benny, sorry. i was wrong. thought this function returned `future<something>` instead of `future<>`. will create a follow up change to correct this.

Reply:
Ah, right, I thought you are saying `co_await` is superfluous, even though you did write `co_return`.

Reply:
> @denesb @bhalevy hi Botond and Benny, sorry. i was wrong. thought this function returned `future<something>` instead of `future<>`. will create a follow up change to correct this.

@tchaikov no need to.  It is not an error to `co_return co_await future<>`, we can live with it.